### PR TITLE
Feature - 3164 - Public Pool Advertisement Page

### LINKED ISSUE #3164: Public Facing Pool Advertisement
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description

A fancy looking version of the pool information, It shows how the information will be perceived by applicants. It is the same URL regardless of the "status" of the pool.

- if pool status is DRAFT/ARCHIVED, link will be visible only to Admin users.
- if pool status is PUBLISHED/CLOSED, link will be visible to everyone.
- if pool status is CLOSED "Apply" buttons will be disabled.
- if pool status is CLOSED "Days remaining" will display closed for submissions message.
- if pool status is PUBLISHED and they hit "Apply" and will be redirected to URL ".../pools/[ID]/apply"

IT level descriptions

### What does IT-01 Technician mean?

```
Technicians (IT-01) provide technical support in the development, implementation, integration, and maintenance of service delivery to clients and stakeholders.

IT Technicians are primarily found in three work streams:
IT Infrastructure Operations
IT Security
IT Software Solutions
```

### What does IT-02 Analyst mean?

`Analysts (IT-02) provide technical services, advice, analysis, and research in their field of expertise to support service delivery to clients and stakeholders. IT analysts are found in all work streams.`

### What does IT-03 Team Lead mean?

```
There are two types of IT-03 employees: those following a management path, and individual contributors.
 Management path: IT Team Leads (IT-03) are responsible for supervising work and project teams for IT services and operations in their field of expertise to support service delivery to clients and stakeholders. IT Team Leads are found in all work streams.
```

### What does IT-03 Technical Advisor mean?

```
There are two types of IT-03 employees: those following a management path, and individual contributors.

Individual contributor: IT Technical Advisors (IT-03) provide specialized technical advice, recommendations and support on solutions and services in their field of expertise in support of service delivery to clients and stakeholders. IT Technical Advisors are found in all work streams.
```

### What does IT-04 Manager mean?

```
There are two types of IT-04 employees: those following a management path, and individual contributors.

Management path: IT Managers (IT-04) are responsible for managing the development and delivery of IT services and/or operations through subordinate team leaders, technical advisors, and project teams, for service delivery to clients and stakeholders. IT Managers are found in all work streams.
```

### What does IT-04 Senior Advisor mean?

```
There are two types of IT-04 employees: those following a management path, and individual contributors.

Individual contributor: IT Senior Advisors provide expert technical advice and strategic direction in their field of expertise in the provision of solutions and services to internal or external clients, and stakeholders. IT Senior Advisors are primarily found in six work streams:

IT Infrastructure Operations
IT Security
IT Software Solutions
IT Database Management
IT Enterprise Architecture
IT Project Portfolio Management
```

## Screenshot: Prototype
image

[Figma prototype](https://www.figma.com/proto/b6lDM86QCJQ6iaAh7qrTYH/Admin-v4-(Direct-Intake)?page-id=0%3A1&node-id=1928%3A3809&viewport=1325%2C4564%2C0.47&scaling=min-zoom&starting-point-node-id=1618%3A6064)

### Closing date message

The closing date message changes styling and copy depending on how close TODAY date is to CLOSING date, here are the rules:

- If TODAY is > CLOSING then the message is: "The deadline for submission has passed"
- or If TODAY is < CLOSING and > (CLOSING - 24 hrs) then: "Closes Today at 11:59 p.m. PST"
- or If TODAY is < CLOSING and > (CLOSING - 48 hrs)  then: "Closes Tomorrow at 11:59 p.m. PST"
- or if TODAY is < CLOSING and > (CLOSING - 72hrs)  then: "Closes in " + n + " days"

image

## Acceptance Criteria
- [x] URL "https://talent.canada.ca/en/admin/pools/[ID]/view" (???)
- [x] Put behind feature flag "Direct Intake"
- [ ] Page needs to behave in different way depending on status (Draft, Published (Active), Closed (Published and Expired)
- [ ] Cypress testing for viewing the page as different users in different states
- [x] Jest tests for the date showing correctly, and for content appearing correctly (eg skills)

### PR DESCRIPTION
**Pull request by esizer**
Resolves #3164 

## Summary

This creates the public facing page for pool advertisements.

## Testing

1. Navigate to `/browse/pools`
2. Click on one of the pools
3. Add `/view` to the end of the path
4. Confirm you can see the page and it matches the prototype

#### Missing Cypress Tests

We are still waiting on mutations to be able to test this with cypress.

#### Status

We have the following statuses and you should expect the following when you encounter them:

- DRAFT
  - 404 if logged in as applicant
  - View normally as admin
- PUBLISHED
  - See as admin or applicant
  - Can apply as applicant
  - Tile with info in header should indicate number of days until it expires
    - Tomorrow and Today are replaced by number of days when appropriate
- EXPIRED
  - See as admin or applicant
  - Tile indicates the post is expired
  - Apply buttons should be disabled

> We will have ARCHIVED and CLOSED eventually but those have not been added yet so are not reflected.

### CODE REVIEW

**Comment by vd1992** [comment]
are the skills going to have something filling them out later? like after mutations are added?

image

**Comment by esizer** [comment]
> are the skills going to have something filling them out later? like after mutations are added?

This is going to be descriptions, we just are not seeding descriptions for skills yet.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/afe3751f01d58e9006b9f67118a754520c03de21/frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx#L363-L367

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+      <span>{children}</span>
+    </h3>
+  );
+};
+
+// NOTE: Not entirely sure why this is failing?
+const accommodationEmail = (chunks: string[]) => (
+  // eslint-disable-next-line jsx-a11y/anchor-has-content
+  <a href="mailto:fames@acanteipsum.ca">{...chunks}</a>
+);
```
Comment (vd1992) [inline frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:86]: I have done linking like this
https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/frontend/talentsearch/src/js/components/roleSalaryForm/RoleSalaryForm.tsx#L93
then 
https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/frontend/talentsearch/src/js/components/roleSalaryForm/RoleSalaryForm.tsx#L263

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+          </TableOfContents.AnchorLink>
+        </TableOfContents.Navigation>
+        <TableOfContents.Content>
+          <TableOfContents.Section id={sections.about.id}>
+            <TableOfContents.Heading>
+              {sections.about.title}
+            </TableOfContents.Heading>
+            <Accordion
+              title={intl.formatMessage({
+                defaultMessage: "What are pool recruitment's?",
```
Comment (vd1992) [inline frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:272]: unneeded apostrophe

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+              {intl.formatMessage({
+                defaultMessage: "Contact and Accommodations",
+                description:
+                  "Title for contact information on pool advertisement",
+              })}
+            </IconTitle>
+            <p>
+              {intl.formatMessage({
+                defaultMessage:
+                  "Do you require accommodations? or do you have any questions about this process?",
```
Comment (vd1992) [inline frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:596]: Capitalize second sentence

**Review by vd1992 — COMMENTED** [review]
a little bit of grammatical nit-picking

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx (hunk 3 above)
Comment (esizer) [reply frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx]: Fixed with [grammar fixes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3338/commits/b69e4fd79319ddc73a9425d9ddbc0b606b1b1509)

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx (hunk 1 above)
Comment (vd1992) [reply frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:86]: 🤷 
the `msg: string` `{msg}` way I linked throws no linting issue so I dunno, maybe the linter doesn't recognize your method

**Review by vd1992 — APPROVED** [review]
- pages load and appear to line up with prototype
- re-seeding database to view different posters, things look right, like time until expiry
- testing in place for important bit at the moment, date testing
- Apply button correctly disabled/enabled based off expiry

not sure if there is anything else I am neglecting
things look good, amazing work!

File: frontend/talentsearch/src/js/components/ClassificationDefinition/ClassificationDefinition.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+interface ClassificationDefinitionProps {
+  name: string;
+}
+
+const definitionMap: Record<string, React.FC> = {
+  [GenericJobTitleKey.TechnicianIt01]: ITDefinitions.LevelOne,
+  [GenericJobTitleKey.AnalystIt02]: ITDefinitions.LevelTwo,
+  [GenericJobTitleKey.TeamLeaderIt03]: ITDefinitions.LevelThreeLead,
+  [GenericJobTitleKey.TechnicalAdvisorIt03]: ITDefinitions.LevelThreeLead,
```
Comment (vd1992) [inline frontend/talentsearch/src/js/components/ClassificationDefinition/ClassificationDefinition.tsx:14]: oops, just noticed this
those are connected to identical things

Comment (esizer) [reply frontend/talentsearch/src/js/components/ClassificationDefinition/ClassificationDefinition.tsx]: Fixed with [fix it-03 advisor defintion](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3338/commits/920abbba3b07d25b7fa179d706e712f20a823381)

Comment (vd1992) [reply frontend/talentsearch/src/js/components/ClassificationDefinition/ClassificationDefinition.tsx]: Nice, think that is all from me

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx [hunk 5]
```diff
@@ -269,7 +269,7 @@ const PoolAdvertisement = ({ poolAdvertisement }: PoolAdvertisementProps) => {
             </TableOfContents.Heading>
             <Accordion
               title={intl.formatMessage({
-                defaultMessage: "What are pool recruitment's?",
+                defaultMessage: "What are pool recruitments?",
                 description:
                   "Title for according describing pool recruitment's",
```
Comment (mnigh) [inline frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:274]: @esizer Don't forget about the description and its fr.json equivalent where the id will change!

**Pull request merged**